src/React.Core/ReactComponent.cs

```diff
@@ -204,39 +236,44 @@ protected virtual void EnsureComponentExists()
 		/// <summary>
 		/// Gets the JavaScript code to initialise the component
 		/// </summary>
-		/// <returns>JavaScript for component initialisation</returns>
-		protected virtual string GetComponentInitialiser()
+		/// <param name="writer">The <see cref="T:System.IO.TextWriter" /> to which the content is written</param>
+		protected virtual void WriteComponentInitialiser(TextWriter writer)
 		{
-			return string.Format(
-				"React.createElement({0}, {1})",
-				ComponentName,
-				_serializedProps
-			);
+			writer.Write("React.createElement(");
+			writer.Write(ComponentName);
+			writer.Write(", ");
+			WriteSerializedProps(writer);
+			writer.Write(')');
 		}
 
 		/// <summary>
-		/// Validates that the specified component name is valid
+		/// Write serialized Props to writer
 		/// </summary>
-		/// <param name="componentName"></param>
-		internal static void EnsureComponentNameValid(string componentName)
+		/// <param name="writer">The <see cref="T:System.IO.TextWriter" /> to which the content is written</param>
+		protected void WriteSerializedProps(TextWriter writer)
 		{
-			var isValid = componentName.Split('.').All(segment => _identifierRegex.IsMatch(segment));
-			if (!isValid)
+			//TODO: cache
```

Comment:
Let's remove any TODOs before merging this in

Reply:
fixed